codigofonte.c

```diff
@@ -1,31 +1,1004 @@
-#include <stdio.h>				//comentar o código faz bem :D
-#include <stdlib.h>
-#include <string.h>
+#include<stdio.h>
+#include<stdlib.h>
+#include<time.h>
+#include<locale.h>
+#include<string.h>
+#include<ctype.h>
+
+
+
+typedef struct endereco ENDERECO;
+struct endereco{
+    char rua[60];
+    char bairro[60];
+    char cidade[60];
+    int numero;
+};
+
+typedef struct cliente CLIENTE;
+struct cliente{
+    char nome[200];
+    char data[11];
+    char predileto[60];
+    int clientevip;
+    int ativo;//1=ativo/0=excluído.
+    ENDERECO end;
+};
+
+typedef struct adm ADM;// STRUCT DE CADASTRO DE USUÁRIO ADM, DO PROGRAMA.
+struct adm
+{
+    char login[20];
+    char senha[20];
+};
+
+
+
+void cadastroCliente()// funçao de cadastro de clientes
+{
+    FILE *arquivo;
+    CLIENTE cll;
+
+    int vip, opmenu, z=3;
+    arquivo = fopen("cliente.dat", "ab");
+    char a[5];
+    char data[11];
+    if(arquivo==NULL)// condição de erro
+	{
+		printf("\nNão foi possível abrir o arquivo, arquivo nulo.");
+		exit(0);
+	}
+    else{
+    do
+    {
+        printf("\t\t\t***********************************\n");
+        printf("\t\t\t\t Cadastrar clientes ");
+        printf("\n\t\t\t***********************************\n");
+
+         printf("Digite o nome do cliente:\n");
+        do
+        {
+            gets(cll.nome);
+            setbuf(stdin,NULL);
```

Comment:
o final da string é '\0' e indica o fim da string, eu nao acho que a função gets come isso, se isso acontecesse a string nao ia ter fim e daria loop infinito. Sim, precisa limpar buffer todas as vezes que usar um gets ou fgets.

uso a setbuf, por causa dessa video aula: https://youtu.be/ixk5RIqABjI